MFVNeutralino/plugins/VertexAuxProducer.cc

```diff
@@ -386,38 +599,35 @@ void MFVVertexAuxProducer::produce(edm::Event& event, const edm::EventSetup& set
       if (verbose) printf("    %i <%f,%f,%f,%f>\n", i, aux.pt[i], aux.eta[i], aux.phi[i], aux.mass[i]);
     }
       
-    //const double sv_r = mag(sv.position().x() - bsx, sv.position().y() - bsy);
-    //const double sv_z = fabs(sv.position().z() - bsz);
-
     auto trkb = sv.tracks_begin();
     auto trke = sv.tracks_end();
+    float sumpt2 = 0;
 
     std::vector<double> costhtkmomvtxdisps;
 
     if (verbose) printf("    tracks %i:\n", int(trke-trkb));
     for (auto trki = trkb; trki != trke; ++trki) {
       const reco::TrackBaseRef& tri = *trki;
+      const reco::TransientTrack sedtri = tt_builder->build(**trki); 
       const reco::TrackRef& trref = tri.castTo<reco::TrackRef>();
       const math::XYZTLorentzVector tri_p4(tri->px(), tri->py(), tri->pz(), tri->p());
+      sumpt2 += pow(tri->pt(),2);
 
-      if (trackicity.count(tri.key()) > 0)
+      if (buffer_trackicity.count({tri.key(), tri.id().id()}) > 0)
         throw cms::Exception("VertexAuxProducer") << "trackicity > 1";
-      else
-        trackicity.insert(tri.key());
+      else 
+        buffer_trackicity.insert({tri.key(), tri.id().id()});
+  
+      // if (trackicity.count(tri.key()) > 0)
+      //   throw cms::Exception("VertexAuxProducer") << "trackicity > 1";
+      // else {
+      //   trackicity.insert(tri.key());
+      // }
```

Comment:
Wasn't this to catch cases where we accidentally add the same track to a vertex twice? @abbywarden and @pkotamnives I feel like you both ran into this at various times, and maybe commented it out at first, and then added it back in once finding the issue? We could double check by uncommenting it, adding some printouts, and running. If it's showing up for the displaced leptons, it could be an odd double counting where a track is considered both a generic track and a lepton track, perhaps.